server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -1130,6 +1131,18 @@ private void fillSegmentInfo(
         segments.put(info.info.name, segment);
     }
 
+    private static KnnVectorsFormat getKnnVectorsFormatForField(Codec codec, String name) {
+        KnnVectorsFormat format;
+        if (codec instanceof Elasticsearch814Codec) {
+            format = ((Elasticsearch814Codec) codec).getKnnVectorsFormatForField(name);
```

Comment:
since `PerFieldMapperCodec` extends `Elasticsearch814Codec` it makes more sense to check on the parent class.